program/llvm-toolchain-ppl.md

```diff
@@ -0,0 +1,252 @@
+
+This project requires the modification of a set of existing public open source code bases maintained as projects of the LLVM Foundation and the REMS project.  Therefore, the processes used within OpenHW will reflect the processes of those upstream projects.
+
+The code base of LLVM is of substantial size:
+
+- Clang: 1.75 MLOC C++;
+- LLVM: 2.1 MLOC C++.
+
+The LLVM code base includes substantial regression test suites (1.5 MLOC including `libcxx` C++ library tests), and success with these test suites is a pre-requisite of upstream acceptance of any patch.
+
+Until CORE-V related contributions are accepted upstream, code will be developed in narrow mirrors of upstream repositories, featuring a single branch based on upstream top of tree.  This will be used as the basis of the patches to be submitted for upstreaming.
+
+<!--- Partners may maintain local backport branches for corrections to official releases used in their internal HW development flows. --->
+
+Once CORE-V contributions are accepted upstream, common CORE-V specific code for LLVM will be developed and maintained in upstream `llvm.org` repositories.  However, dedicated extensions may continue to be developed in member-owned repositories tracking the upstream repository.
+
+### Summary timeline
+
+- Preliminary Project Launch (PPL, this document): TBD
+  - includes project plan
+
+- Project Launch (PL): TBD
+- First OHG release (CVA6 64/32 bits, HW verification equivalence to GCC): TBD
+- Upstreaming of first release: 2021 H1, targeting LLVM 13.0.
+
+The schedule of LLVM releases follows a six-month cycle which additionally depends on the progress of on-going releases.  This makes the scheduling of upstream contributions more flexible (and also more difficult to schedule upfront).
+
+## OpenHW members/participants committed to participate in this project
+
+All OpenHW Group members are invited to contributed expertise to this project. At present we are aware of the interest and commitment of:
+
+1. Thales
+2. Embecosm
+
+## Project Manager and technical lead
+
+- TBD
+
+This is a much lighter role than a traditional project manager, since the requirements are fully defined through the hardware design process.  The primary roles will be:
+ tracking and reporting status to OpenHW Software TG and Technical WG;
+- identifying and resolving issues related to the outcome.
+
+## Project documents
+
+The following project documents will be created:
+- Preliminary Project Proposal (this document)
+
+  - including an initial project plan
+  - including an initial risk register
+
+- Project Proposal, an updated version of this document
+
+  - separate detailed project plan
+  - separate risk register
+
+## Virtual customer
+
+It is proposed that the OpenHW Group Cores TWG acts as a "virtual customer" to exercise the compiler as it is developed.  In particular, the CVA6 project will exercise the multi-XLEN and core functionality of the LLVM compiler in the CVA6 verification flow.
```

Comment:
I think you mean the HW TG as customer - they can try out LLVM on the FPGA platform.